.github/workflows/pr.yml

```diff
@@ -13,31 +13,35 @@ jobs:
     permissions:
       pull-requests: write
     steps:
-    - uses: actions/github-script@v7
-      with:
-        script: |
-          const pr = await github.rest.pulls.get({
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            pull_number: context.issue.number
-          })
-          const isTitleValid = /^\[#\d+\] /.test(pr.data.title)
-          const isDescriptionValid = /([Ff]ix(es|ed)?|[Cc]lose(s|d)?|[Rr]esolve(s|d)?|[Pp]art [Oo]f) #\d+/.test(pr.data.body)
-          if (isTitleValid && isDescriptionValid) {
-            return
-          }
-          let body = `Hi @${pr.data.user.login}, thank you for your interest in contributing to TEAMMATES!
-            However, your PR does not appear to follow our [contribution guidelines](https://teammates.github.io/teammates/process.html#step-4-submit-a-pr):\n\n`
-          if (!isTitleValid) {
-            body += "- Title must start with the issue number the PR is fixing in square brackets, e.g. `[#<issue-number>]`\n"
-          }
-          if (!isDescriptionValid) {
-            body += "- Description must reference the issue number the PR is fixing, e.g. `Fixes #<issue-number>` (or `Part of #<issue-number>` if the PR does not address the issue fully)\n"
-          }
-          body += "\nPlease address the above before we proceed to review your PR."
-          await github.rest.issues.createComment({
-            issue_number: context.issue.number,
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            body,
-          })
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const pr = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            })
+            const isIssueOpen = pr.data.state.description === 'open';
+            const isTitleValid = /^\[#\d+\] /.test(pr.data.title)
+            const isDescriptionValid = /([Ff]ix(es|ed)?|[Cc]lose(s|d)?|[Rr]esolve(s|d)?|[Pp]art [Oo]f) #\d+/.test(pr.data.body)
+            if (isTitleValid && isDescriptionValid && isIssueOpen) {
+              return
+            }
+            let body = `Hi @${pr.data.user.login}, thank you for your interest in contributing to TEAMMATES!
+              However, your PR does not appear to follow our [contribution guidelines](https://teammates.github.io/teammates/process.html#step-4-submit-a-pr):\n\n`
+            if (!isTitleValid) {
+              body += "- Title must start with the issue number the PR is fixing in square brackets, e.g. `[#<issue-number>]`\n"
+            }
+            if (!isDescriptionValid) {
+              body += "- Description must reference the issue number the PR is fixing, e.g. `Fixes #<issue-number>` (or `Part of #<issue-number>` if the PR does not address the issue fully)\n"
+            }
+            if (!isIssueOpen) {
```

Comment:
Based on this branching, it is possible that the body contains the message for not having a `Fixes #<issue-number>` as well as the message for not addressing an open issue right? I would assume logic should work  something like this:
- If isDescriptionValid is false, it means the PR description doesn't contain a valid reference to an issue number.
- If isIssueOpen is checking whether the referenced issue is open (assuming it was implemented correctly), it wouldn't be able to perform this check if there's no valid issue reference in the description.

Based on the above, a more logical flow should look something like:

- Check if the description is valid (contains a proper issue reference)
- Only if the description is valid, then fetch and check the state of the referenced issue (I believe the issue number can be extracted using capture groups)

To use a capture group, the extraction process might look something like this (this is just an example, do test it out and verify)

```
const descriptionRegex = /(?:[Ff]ix(?:es|ed)?|[Cc]lose(?:s|d)?|[Rr]esolve(?:s|d)?|[Pp]art [Oo]f) #(\d+)/;
const extractIssueNum = (description) => {
  const match = description.match(descriptionRegex);
  return match ? match[1] : null;
};
const issueNumber = extractIssueNumber(pr.data.body);
```